src/components/admin/form-components/searchables.jsx

```diff
@@ -26,3 +27,7 @@ export const SearchableArticlesWithPubDate = withFalcorSearch(
   },
   'articles',
 )(SearchBar);
+
+export const SearchableAuthorsSelector = withSelector(
```

Comment:
I still don't know if its worth adding a dependency for two functions which are syntactic sugar around one liners(Together, both `compose` and `getDisplayName` come in at less than 20 lines and most of it is making sure it returns correctly when called needlessly), especially when one of those functions hasn't been updated in over a year and the other one hasn't been updated in 2 years. Not saying recompose isn't worth it, but `compose` and `getDisplayName` don't strike me as a compelling argument.

Reply:
```js
const getDisplayName = Component => {
  if (typeof Component === 'string') {
    return Component
  }

  if (!Component) {
    return undefined
  }

  return Component.displayName || Component.name || 'Component'
}

export default getDisplayName
```
```js
export default function compose(...funcs) {
  if (funcs.length === 0) {
    return arg => arg
  }

  if (funcs.length === 1) {
    return funcs[0]
  }

  return funcs.reduce((a, b) => (...args) => a(b(...args)))
}
```

Reply:
We would be importing a 15 kb(by comparison, react as a whole is only 5 kb) package for 20 lines of code. Seems overkill to me.

Reply:
Is recompose really 15kb? Okay that I didn't know, I imagined it would be a few 100 bytes. I was gonna link you to this though https://github.com/sindresorhus/ama/issues/10#issuecomment-117766328
but the 15kb may be a good argument

Reply:
Recompose is a library intended to do much fancier things with functional programming and react. While importing one line packages is definitely something done in JS land, doing so carelessly is what leads to package bloat and https://www.theregister.co.uk/2016/03/23/npm_left_pad_chaos/

Reply:
https://medium.com/@jdan/i-peeked-into-my-node-modules-directory-and-you-wont-believe-what-happened-next-b89f63d21558

Read this and I think you may see that while one-line imports have potential to be nice, they rarely in practice are!

Reply:
(the above is a joke to be clear hehe, I just hate unneeded imports)

Reply:
Oh lol it was a joke? :P, hahahaha, I didn't get that, now I understand though, because I found the PR for Guy Fieri, but I guess it was after the article was published and they were joking haha. They apparently had it merged in in between releases before removing it again hahaha, that's a pretty good sense of humor.

And yeah I know of the left pad incident, but if you go too far down that rabbit hole you just end up reinventing everything. I'll read your SO link now, but no matter what I think I agree that adding 15 kb (as long as we don't have Webpack tree shaking setup properly anyway, though we should actually maybe consider using [this plugin](https://github.com/acdlite/recompose#using-babel-plugin-lodash) as it seems it would also drastically reduce the bundle size added by lodash) for two very simple functions is overkill, especially when the first one has nothing to do with production code but just dev tools

Reply:
Also, I'm definitely also aware of not wanting too many dependencies, but I guess I'm further in the other direction than you, (also this is just a tangent conversation by now, I agreed to leave it as is), did you read the comments to the Engineering Manager at SO's post? Quite a few people disagree with him ;)